crates/codegen/schema/src/types/production.rs

```diff
@@ -0,0 +1,91 @@
+use std::rc::Rc;
+
+use indexmap::IndexMap;
+use schemars::JsonSchema;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+
+use super::{parser::Parser, precedence_parser::PrecedenceParser, scanner::Scanner};
+
+pub type ProductionRef = Rc<Production>;
+
+#[derive(Deserialize, Serialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields)]
+#[serde(tag = "kind")]
+pub enum Production {
+    Scanner {
+        name: String,
+        #[serde(flatten)]
+        version_map: VersionMap<Scanner>,
+    },
+    TriviaParser {
```

Comment:
Thoughts on renaming `Production::TriviaParser` to just `Production::Trivia`, and `Production::PrecedenceParser` to  `Production::Operators`?
Not only for brevity, but to differentiate between them and `Production::Parser`, which has different semantics.

Reply:
Will create an issue for later

Reply:
#373